docs: add release notes for 0.29.1 (#9014)
determined-ai/determined: docs/release-notes.rst

 Version 0.29
**************

Version 0.29.1
==============

**Release Date:** March 18, 2024

**New Features**

-  Include early-access NVIDIA NGC-based images in our environment offerings. These images are

>    accessible from `pytorch-ngc <https://hub.docker.com/r/determinedai/pytorch-ngc>`_ or
>    `tensorflow-ngc <https://hub.docker.com/r/determinedai/tensorflow-ngc>`_. By downloading and

Reviewer: ```suggestion
   accessible from `pytorch-ngc <https://hub.docker.com/r/determinedai/pytorch-ngc>`__ or
   `tensorflow-ngc <https://hub.docker.com/r/determinedai/tensorflow-ngc>`__. By downloading and
```

   using these images, users acknowledge and agree to the terms and conditions of all third-party
   software licenses contained within, including the `NVIDIA Deep Learning Container License
   <https://developer.download.nvidia.com/licenses/NVIDIA_Deep_Learning_Container_License.pdf>`__.

>    Users can build their own images from a specified NGC container version by using the

Reviewer: Had this before, seems to have gotten lost:
```suggestion
   Users can build their own images from a specified NGC container version using the
```
If you don't like it, that's fine; it doesn't matter much. Just pointing out in case it was unintentional.

Author: Thanks, this was unintentional

>    Users can build their own images from a specified NGC container version by using the
>    ``build-pytorch-ngc`` or ``build-tensorflow-ngc`` workflows located in our environments
>    ``MakeFile`` in the `environments repository <https://github.com/determined-ai/environments>`_.

Reviewer: ```suggestion
   Users can build their own images from a specified NGC container version using the
   ``build-pytorch-ngc`` or ``build-tensorflow-ngc`` targets in the makefile in our `environments
   repository <https://github.com/determined-ai/environments>`__.
```

**Improvements**

> -  Eliminate TensorFlow 2.8 images from our offerings. Default TensorFlow 2.11 images remain

Reviewer: ```suggestion
-  Images: Eliminate TensorFlow 2.8 images from our offerings. Default TensorFlow 2.11 images remain
```

   available for TensorFlow users.

**Bug Fixes**

> -  Experiments: Fix an issue where experiments in the ``STOPPING_CANCELED`` state on master restart,

Reviewer: ```suggestion
-  Experiments: Fix an issue where experiments in the ``STOPPING_CANCELED`` state on master restart
```
